mart/attack/adversary.py

```diff
@@ -81,18 +84,19 @@ class Attacker(AttackerCallbackHookMixin, torch.nn.Module):
     def __init__(
         self,
         *,
-        perturber: BatchPerturber | Perturber,
+        perturber: Perturber,
```

Comment:
We should figure out some way to exclude adversary parameters in model checkpoint: e.g. `model.input_adv_training.attacker.perturber.perturbation`.

We won't be able to resume adversarial training with extra adversary parameters, unless we set `strict=False`. But that could hide other issues.

Reply:
Isn't that the job of the Adversary? Sometimes I actually want to save the perturbation like in Universal attacks.

Reply:
Maybe we should create a separate mechanism to export the universal adversarial perturbation?

You are right about the two tests. I would change `tests/test_adversary.py::test_hidden_params_after_forward` so that after `forward()`, the adversary module still does not have the parameter, because we don't want it to be exported to the model checkpoint.

Reply:
I added a PR to hide the adversarial params
#150